src/ucp/api/ucp.h

```diff
@@ -2401,6 +2540,95 @@ ucs_status_ptr_t ucp_am_send_nb(ucp_ep_h ep, uint16_t id,
                                 ucp_send_callback_t cb, unsigned flags);
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Send Active Message.
+ *
+ * This routine sends an Active Message to an ep. If the operation completes
+ * immediately, then the routine returns NULL and the call-back function is
+ * ignored, even if specified. Otherwise, if no error is reported and a callback
+ * is requested (i.e. the UCP_OP_ATTR_FIELD_CALLBACK flag is set in the
+ * op_attr_mask field of @a param), then the UCP library will schedule
+ * invocation of the call-back routine @a param->cb.send upon completion of the
+ * operation.
+ *
+ * @note If UCP_OP_ATTR_FLAG_NO_IMM_CMPL flag is set in the op_attr_mask field
+ *       of @a param, then the operation will return a request handle, even if
+ *       it completes immediately.
+ * @note This operation does not support CUDA memory.
+ * @note This operation supports specific flags, which can be passed
+ *       in @a param by @ref ucp_request_param_t.flags. The exact set of flags
+ *       is defined by @ref ucp_send_am_flags.
+ *
+ * @param [in]  ep            UCP endpoint where the Active Message will be run.
+ * @param [in]  id            Active Message id. Specifies which registered
+ *                            callback to run.
+ * @param [in]  header        Active message header.
+ * @param [in]  header_length Active message header length in bytes.
+ * @param [in]  buffer        Pointer to the data to be sent to the target node
+ *                            of the Active Message.
+ * @param [in]  count         Number of elements to send.
```

Comment:
Seems like we need datatype here

Reply:
it is inside ucp_request_param_t, which is a common argument for all `_nbx` APIs now